app/Controllers/ProtectedUploadController.php

```
@@ -0,0 +1,196 @@
+
+        $length = $end - $start + 1;
+        $headers['Content-Length'] = (string) $length;
+
+        $body = $this->buildBody($handle, $start, $end, $size);
+
+        $res = new SlimResponse($status, null, $body);
+        foreach ($headers as $name => $value) {
+            $res = $res->withHeader($name, $value);
+        }
+        return $res;
+    }
```

PHPMD (2.15.0)

[warning] The method serve() has a Cyclomatic Complexity of 25. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)